extensions/amp-analytics/0.1/test/test-amp-analytics.js

```diff
@@ -1203,6 +1203,148 @@ describe('amp-analytics', function() {
     });
   });
 
+  describe('enabled', () => {
+    function getConfig() {
+      return {
+        'requests': {
+          'pageview1': '/test1=${requestCount}',
+        },
+        'triggers': {
+          'conditional': {
+            'on': 'visible',
+            'request': 'pageview1',
+            'vars': {},
+          },
+        },
+        'vars': {},
+      };
+    }
+
+    it('allows a request through', () => {
+      const analytics = getAnalyticsTag(getConfig());
+
+      return waitForSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.be.calledOnce;
+      });
+    });
+
+    it('allows a request based on a variable', () => {
+      const config = getConfig();
+      config.triggers.conditional.enabled = '${foo}';
+      config.triggers.conditional.vars.foo = 'bar';
+      const analytics = getAnalyticsTag(config);
+
+      return waitForSendRequest(analytics).then(() => {
+        expect(sendRequestSpy).to.be.calledOnce;
+      });
+    });
+
+    it('allows a request based on url-replacements', () => {
+      const config = getConfig();
+      config.triggers.conditional.enabled = '${pageViewId}';
+      const analytics = getAnalyticsTag(config);
+
+      const urlReplacements = urlReplacementsForDoc(analytics.element);
+      sandbox.stub(urlReplacements.getVariableSource(), 'get').returns(0);
```

Comment:
The result of expandUrlAsync  is always a string because it gets encoded every time, that's the reason 0 (that gets translated to '0') is treated as true